### ISSUE TITLE
try MiniTensor (SVD for polar, and eigen)

### ISSUE THREAD

**Issue by ibaned** [ibaned/omega_h#86]

**Comment by ibaned** [comment]
Below is a copy-paste of a report on issues for Alexa. Whether Alexa decides to spend time fixing these or not, I don't intend to use MiniTensor in Omega_h until they are resolved.

1. `sizeof(Tensor<double,3>) == 96`, and `sizeof(double)*9==72`.
    `96-72=24=8*3`, meaning a `Tensor` has three extra words we don't need.
    After digging through the code, I have identified what these three words are:
 - A "size" integer, pointless because its always = 3
 - An "order" integer, pointless because its always = 2
 - A `vtable` pointer, because `TensorBase` is virtual ! (this is dangerous on the GPU)

    These are here because `Tensor` (which really means 2nd-order tensor) derives from `TensorBase`, which has the capability     to handle changing both tensor size and order *at runtime*.

2. Non-GPU-capable functions are being called at the GPU level, specifically `boost::tie`, `std::pair`, and `Teuchos::ScalarTraits`

3. All constructors of a `Tensor` will fill its components with `NaN`. This can get really expensive (I've tried this in Omega_h, slowed it down a lot), and theres currently no way to turn it off. Also, You're not supposed to put something in `Kokkos::View` that has a non-trivial default constructor.

Basically, I think MiniTensor needs a lot of work in order to be correct and memory+runtime efficient on the GPU, at least to the level that `Omega_h::Matrix` already is.

Alejandro and I are discussing solutions to all the above problems, but it will take some work. The calling of non-GPU functions on the GPU is especially troublesome.

**Comment by aprokop** [comment]
Hey @ibaned, I needed SVD for some other project, so I rewrote the MiniTensor version with something that runs on GPUs. It's not optimized yet (and I hope that somebody else would write a proper batched version) but if you are interested, it's [here](https://github.com/aprokop/kokkos_perf/blob/master/svd/svd.cpp).

**Comment by ibaned** [comment]
Thanks @aprokop , thats much appreciated!
